org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/javaeditor/codemining/CalleeJavaMethodParameterVisitor.java

```diff
@@ -104,25 +104,34 @@ public boolean visit(MethodInvocation methodInvocation) {
 			IMethodBinding methodBinding= methodInvocation.resolveMethodBinding();
 			if (methodBinding != null) {
 				IMethod method = resolveMethodBinding(methodBinding);
-				collectParameterNamesCodeMinings(method, arguments, methodBinding.isVarargs());
+				collectParameterNamesCodeMinings(method, arguments, methodBinding);
 			}
 		}
 		return super.visit(methodInvocation);
 	}
 
-	protected void collectParameterNamesCodeMinings(IMethod method, List<?> arguments, boolean isVarargs) {
+	protected void collectParameterNamesCodeMinings(IMethod method, List<?> arguments, IMethodBinding mbinding) {
```

Comment:
Also, we have a method binding != null at each call site. Why do we even bother with computing the `IMethod` which may be `!= null` if we already have an object that can serve just as well?

Reply:
You're right. I think with this API addition from jdt.core, there's no need for `IMethod` at all. We need the method binding for `isVarArgs()` so if that needs to exist, we don't need to bother also getting the parameter names. I'll make the change and see if it things work as expected.

Reply:
I've tried eliminating `IMethod#getParameterNames()` and just using the `IMethodBinding#getParameterNames()` but it doesn't seem to work in some cases. It breaks [`ParameterNamesCodeMiningTest`](https://github.com/eclipse-jdt/eclipse.jdt.ui/blob/master/org.eclipse.jdt.text.tests/src/org/eclipse/jdt/text/tests/codemining/ParameterNamesCodeMiningTest.java). Looks to be empty for certain testcases, while `IMethod@getParameterNames()` does work. No idea why yet.

On the plus side, I didn't think this had testcases, so may as well add one for records to this PR.